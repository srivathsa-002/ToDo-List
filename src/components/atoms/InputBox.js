import React from 'react';
import "../../styles/components/widget.css";

export default class InputBox extends React.Component {
    render() {
        return (
            <input
              data-testid="input-field"
              className="widget-header-input"
              name="text"
              value={this.props.value}
              onChange={this.props.onChange}
              placeholder="Todo..."
            />
        );
    }
}