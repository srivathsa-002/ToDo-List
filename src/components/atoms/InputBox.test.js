import React from "react";
import {render,fireEvent} from "@testing-library/react";
import InputBox from "./InputBox";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("InputBox testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<InputBox />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId, queryByPlaceholderText} = render(<InputBox />)

    expect(getByTestId("input-field")).toBeTruthy()
    expect(queryByPlaceholderText("Todo...")).toBeTruthy()
})

describe("Input value", () => {
    it("updates on change", () => {
        const {queryByPlaceholderText} = render(<InputBox />)
        const inputBox = queryByPlaceholderText("Todo...")
        fireEvent.change(inputBox, {target: {value: "test"}})
        expect(inputBox.value).toBe("test")
    })
})
