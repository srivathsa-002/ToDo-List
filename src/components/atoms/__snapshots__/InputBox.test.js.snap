// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox testing Matches the snapshot 1`] = `
<input
  className="widget-header-input"
  data-testid="input-field"
  name="text"
  placeholder="Todo..."
/>
`;
